src/vm/eventtracepriv.h

```diff
@@ -307,10 +312,42 @@ class BulkTypeEventLogger
 public:
     BulkTypeEventLogger() :
         m_nBulkTypeValueCount(0),
-        m_nBulkTypeValueByteCount(0)
+        m_nBulkTypeValueByteCount(0),
+#ifdef FEATURE_PAL
+        m_pBulkTypeEventBuffer(NULL)
+#endif
+    {
+        CONTRACTL
+        {
+            NOTHROW;
+            GC_NOTRIGGER;
+            MODE_ANY;
+        }
+        CONTRACTL_END;
+
+#ifdef FEATURE_PAL
+        m_pBulkTypeEventBuffer = new (nothrow) BYTE[kSizeOfEventBuffer];
+#endif
+    }
+
+#ifdef FEATURE_PAL
+    ~BulkTypeEventLogger()
     {
-        LIMITED_METHOD_CONTRACT;
+        CONTRACTL
+        {
+            NOTHROW;
+            GC_NOTRIGGER;
+            MODE_ANY;
+        }
+        CONTRACTL_END;
+
+        if(m_pBulkTypeEventBuffer != NULL)
```

Comment:
A nit - this check is not necessary, the delete operator is required to handle the case when the pointer is null. From the C++ doc at http://en.cppreference.com/w/cpp/memory/new/operator_delete:
> In all cases, if ptr is a null pointer, the standard library deallocation functions do nothing

Reply:
Thanks @janvorli.  Fixed.